controller/webhook/config.go

```diff
@@ -0,0 +1,81 @@
+package webhook
+
+import (
+	"bytes"
+	"encoding/base64"
+	"html/template"
+
+	"github.com/linkerd/linkerd2/controller/k8s"
+	"github.com/linkerd/linkerd2/pkg/tls"
+	log "github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ConfigOps declares the methods used to manage the webhook configs in the cluster
+type ConfigOps interface {
+	Create(*k8s.API, *bytes.Buffer) (string, error)
+	Get(*k8s.API) error
+	Delete(*k8s.API) error
+}
+
+// Config contains all the necessary data to build and persist the webhook resource
+type Config struct {
+	MetricsPort         uint32
+	WebhookConfigName   string
+	WebhookServiceName  string
+	TemplateStr         string
+	Ops                 ConfigOps
+	Handler             handlerFunc
+	api                 *k8s.API
+	controllerNamespace string
+	rootCA              *tls.CA
+}
+
+// Create deletes the webhook config if it already exists and then creates
+// a new one
+func (c *Config) Create() (string, error) {
+	exists, err := c.Exists()
+	if err != nil {
+		return "", err
+	}
+
+	if exists {
+		log.Info("deleting existing webhook configuration")
+		if err := c.Ops.Delete(c.api); err != nil {
```

Comment:
consider passing in `c.api.Client.AdmissionregistrationV1beta1()` for a more specific interface?